pages/client-libraries/java.mdx

```diff
@@ -4,105 +4,687 @@ description: Peruse our guide on implementing Java in Memgraph. Comprehensive do
+ - `run` - executes a implicit auto-commit transaction and returns result.
+
+
+<Callout type="warning">
+
+Error with multicommand transactions
+
+If you experience `Index manipulation not allowed in multicommand transactions` error,  use [implicit (or auto-commit) transactions](#implicit-transactions) instead.
+
+</Callout>
+
+##### Managed transactions
+
+Let's take a look into the basic example if you want to read data from the database.
+Now you can run simple transaction via `writeTransaction` like this: 
+
+```java
+public void createNode() {
+    try (Session session = driver.session(SessionConfig.forDatabase("memgraph"))) {
+        session.writeTransaction(tx -> {
+            tx.run("CREATE (n:Technology {name:'Memgraph'})");
+            tx.run("CREATE (n:Technology {name:'Java'})");
+            //Some other queries 
+            return null;
+        });
+    }
+}
+```
+As you can see previus examples from guide has used the simple transaction management, it alows you to run multiple queries in a single transaction and have a control over the transaction. You could have multiple logic blocks in the transaction, rollback or end the transaction if needed. 
+Also mange transaction have a retry logic, which means that if the transaction fails, it will be retried. Managed transactions are the most flexible way to run a Cypher query.
+
+##### Implicit transactions 
+
+Implicit auto commit queries are the simplest way to run a Cypher query since they won't be automatically retried as with writeTransaction and readTransaction. 
+With implicit auto commit transactions, you don't have the same control of transaction as with manged explicit transactions. 
+
+Here is the example of implicit transaction:
+```java
+public void createNode() {
+    try (Session session = driver.session(SessionConfig.forDatabase("memgraph"))) {
+        session.run("CREATE (n:Technology {name:'Memgraph'})");
+    }
+}
+```
+
+As you can see in this case you can run a simple query without any transaction management. This simplifys the experience but you lose some flexibility when running the queries. 
+Since explicit transaction are multi-statement transactions, creating an index, constraints, auth config queries are not possible in multi-line transactions. Hence, you need to use implicit transactions for those queries.
+
+#### Asynchronous transaction managment
+
+##### Session management
+
+Asynchronous sessions provide a API to work with futures. This allows client applications to work within asynchronous code bases. Asynchronous sessions are not thread safe, so you should create a new session for each thread.
+Session lifetime begins with session construction, session is typicaly closed automatically when all results are consumed. 
+
+The session is opened with the following code snippet: 
+
+```java
+ var session = driver.session(AsyncSession.class);
+```
+
+Via async session API you can use following transaction management methods:
```

Comment:
```suggestion
Asynchronous sessions provide an API to work with futures. This allows client applications to work within asynchronous code bases. Asynchronous sessions are not thread-safe, so you should create a new session for each thread.
Session lifetime begins with session construction, and the session is typically closed automatically when all results are consumed. 

The session is opened using the following code snippet: 

```java
 var session = driver.session(AsyncSession.class);
```

You can use the following transaction management methods via async session API.
```

to work with futures? What does that mean?